backend/vxlan/vxlan_network.go

```diff
@@ -111,140 +85,59 @@ type vxlanLeaseAttrs struct {
-	}
+		switch event.Type {
+		case subnet.EventAdded:
+			log.V(2).Infof("Adding subnet: %s PublicIP: %s VtepMAC: %s", event.Lease.Subnet, event.Lease.Attrs.PublicIP, net.HardwareAddr(attrs.VtepMAC))
+			if err := nw.dev.AddArp(neighbor{IP: event.Lease.Subnet.IP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("AddArp failed: ", err)
+			}
 
-	// Loop over the existing FDB entries, deleting any that shouldn't be there
-	for j, marker := range fdbEntryMarker {
-		if !marker && fdbTable[j].IP != nil {
-			err := nw.dev.DelL2(neighbor{IP: ip.FromIP(fdbTable[j].IP), MAC: fdbTable[j].HardwareAddr})
-			if err != nil {
-				log.Error("Delete L2 failed: ", err)
+			if err := nw.dev.AddFdb(neighbor{IP: event.Lease.Attrs.PublicIP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("AddFdb failed: ", err)
 			}
-		}
-	}
 
-	// Loop over the events (skipping marked ones), adding them to the FDB table.
-	for i, marker := range eventMarker {
-		if !marker {
-			err := nw.dev.AddL2(neighbor{IP: batch[i].Lease.Attrs.PublicIP, MAC: net.HardwareAddr(leaseAttrsList[i].VtepMAC)})
-			if err != nil {
-				log.Error("Add L2 failed: ", err)
+			// Set the route - the kernel would ARP for the Gw IP address if it hadn't already been set above so make sure
+			// this is done last.
+			if err := netlink.RouteReplace(&route); err != nil {
+				log.Errorf("failed to add route (%s -> %s): %v", route.Dst, route.Gw, err)
 			}
-		}
-	}
-	return nil
-}
 
-func (nw *network) handleMiss(miss *netlink.Neigh) {
-	switch {
-	case len(miss.IP) == 0 && len(miss.HardwareAddr) == 0:
-		log.V(2).Info("Ignoring nil miss")
+		case subnet.EventRemoved:
+			log.V(2).Infof("Removing subnet: %s PublicIP: %s VtepMAC: %s", event.Lease.Subnet, event.Lease.Attrs.PublicIP, net.HardwareAddr(attrs.VtepMAC))
 
-	case len(miss.HardwareAddr) == 0:
-		nw.handleL3Miss(miss)
+			if err := nw.dev.DelArp(neighbor{IP: event.Lease.Subnet.IP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("DelArp failed: ", err)
+			}
 
-	default:
-		log.V(4).Infof("Ignoring not a miss: %v, %v", miss.HardwareAddr, miss.IP)
-	}
-}
+			if err := nw.dev.DelFdb(neighbor{IP: event.Lease.Attrs.PublicIP, MAC: net.HardwareAddr(attrs.VtepMAC)}); err != nil {
+				log.Error("DelFdb failed: ", err)
+			}
 
-func (nw *network) handleL3Miss(miss *netlink.Neigh) {
-	route := nw.routes.findByNetwork(ip.FromIP(miss.IP))
-	if route == nil {
-		log.V(0).Infof("L3 miss but route for %v not found", miss.IP)
-		return
-	}
+			if err := netlink.RouteDel(&route); err != nil {
+				log.Errorf("failed to delete route (%s -> %s): %v", route.Dst, route.Gw, err)
+			}
 
-	if err := nw.dev.AddL3(neighbor{IP: ip.FromIP(miss.IP), MAC: route.vtepMAC}); err != nil {
-		log.Errorf("AddL3 failed: %v", err)
-	} else {
-		log.V(2).Infof("L3 miss: AddL3 for %s succeeded", miss.IP)
+		default:
+			log.Error("Internal error: unknown event type: ", int(event.Type))
```

Comment:
any reason why you have some error logs with `log.Error` and uppercase error message vs `log.Errorf` and a lowercase message?